paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -162,6 +207,40 @@ internal class CustomerSheetViewModel(
                 loadCustomerSheetState()
             }
         }
+
+        viewModelScope.launch {
+            selectPaymentMethodState.collectLatest { selectPaymentMethodState ->
+                updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
+                    selectPaymentMethodState
+                }
+            }
+        }
+
+        viewModelScope.launch {
+            viewState.collectLatest { viewState ->
+                if (
+                    viewState is CustomerSheetViewState.SelectPaymentMethod &&
+                    !customerState.value.canShowSavedPaymentMethods
+                ) {
+                    delay(REMOVAL_TRANSITION_DELAY)
```

Comment:
This `delay` is to allow the `SelectPaymentMethod` screen to show the last payment method removed before transitioning to the add screen.